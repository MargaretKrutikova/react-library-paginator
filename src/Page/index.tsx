import * as React from 'react';
import * as PropTypes from 'prop-types';
import { PageStyles, PageClasses } from '../types';
import classNames from 'classnames';

import styles from '../styles.css';

export type PropType = {
  isActive?: boolean;
  isDisabled: boolean;
  page: number;
  onPageChange: (page: number) => void;
  useBootstrapClasses: boolean;
} & DefaultProps;

type DefaultProps = {
  styles: PageStyles;
  classes: PageClasses;
};

class Page extends React.PureComponent<PropType> {
  static defaultProps: DefaultProps = {
    classes: {},
    styles: {}
  };
  static propTypes = {
    isActive: PropTypes.bool,
    isDisabled: PropTypes.bool.isRequired,
    page: PropTypes.number.isRequired,
    onPageChange: PropTypes.func.isRequired
  };
  setPage = (event: React.MouseEvent<HTMLElement>, page: number) => {
    if (event) {
      event.preventDefault();
    }
    if (!this.props.isDisabled && !this.props.isActive) {
      this.props.onPageChange(page);
    }
  };
  render() {
    const { page, isActive, isDisabled, useBootstrapClasses } = this.props;
    const {
      pageItem,
      pageLink,
      pageLinkActive,
      pageLinkDisabled
    } = this.props.classes;

    const customStyles = this.props.styles;

    const pageLinkStyles = {
      ...customStyles.pageLink,
      ...(isActive ? customStyles.pageLinkActive : {}),
      ...(isDisabled ? customStyles.pageLinkDisabled : {}),
      ...(useBootstrapClasses && { boxShadow: 'none' })
    };

    return (
      <li
        className={classNames(
          useBootstrapClasses && {
            'page-item': true,
            active: isActive,
            disabled: isDisabled
          },
          [pageItem]
        )}
        style={customStyles.pageItem}
      >
        <a
          href="#"
          className={classNames(
            styles.rlPageLinkBase,
            pageLink,
            !useBootstrapClasses
              ? {
                  [styles.rlPageLinkBaseActive]: isActive,
                  [styles.rlPageLinkBaseDisabled]: isDisabled,
                  [pageLink || styles.rlPageLink]: true,
                  [pageLinkActive || styles.rlPageLinkActive]: isActive,
                  [pageLinkDisabled || styles.rlPageLinkDisable]: isDisabled
                }
              : 'page-link'
          )}
          style={pageLinkStyles}
          onClick={event => this.setPage(event, page)}
        >
          {this.props.children || page}
        </a>
      </li>
    );
  }
}

export default Page;
