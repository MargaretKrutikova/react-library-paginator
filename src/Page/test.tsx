import * as React from 'react';
import * as Enzyme from 'enzyme';
import Page, { PropType } from '.';

describe('Page', () => {
  it('it changes classes if isActive or isDisabled props change', () => {
    const testProps: PropType = {
      useBootstrapClasses: false,
      isActive: false,
      isDisabled: false,
      page: 4,
      // tslint:disable-next-line:no-empty
      onPageChange: (_: number) => {}
    };
    const component: Enzyme.ShallowWrapper = Enzyme.shallow(
      <Page {...testProps} />
    );

    expect(component).toMatchSnapshot();

    // change props and rerender
    component.setProps({ isActive: true });
    expect(component).toMatchSnapshot();

    // change props and rerender
    component.setProps({ isActive: false });
    component.setProps({ isDisabled: true });
    expect(component).toMatchSnapshot();
  });

  it(' calls onPageChange with current page if page is not disabled and not active', () => {
    const mockCallback = jest.fn();
    const testProps: PropType = {
      useBootstrapClasses: false,
      isActive: false,
      isDisabled: false,
      page: 4,
      onPageChange: (page: number) => mockCallback(page)
    };
    const component: Enzyme.ShallowWrapper = Enzyme.shallow(
      <Page {...testProps} />
    );

    component.find('a').simulate('click');

    expect(mockCallback.mock.calls.length).toBe(1);
    expect(mockCallback.mock.calls[0][0]).toBe(4);
  });

  it(" doesn't call onPageChange if page is disabled or active", () => {
    const mockCallback = jest.fn();
    const testProps: PropType = {
      useBootstrapClasses: false,
      isActive: true,
      isDisabled: false,
      page: 4,
      onPageChange: (page: number) => mockCallback(page)
    };
    const component: Enzyme.ShallowWrapper = Enzyme.shallow(
      <Page {...testProps} />
    );

    component.find('a').simulate('click');
    expect(mockCallback.mock.calls.length).toBe(0);

    component.setProps({ isActive: false, isDisabled: true });
    component.find('a').simulate('click');
    expect(mockCallback.mock.calls.length).toBe(0);
  });
});
