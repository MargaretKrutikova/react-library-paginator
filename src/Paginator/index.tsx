import * as React from "react";
import * as PropTypes from "prop-types";
import Page from "../Page";

type PropType = {
  currentPage: number;
  totalPages: number;
  pagesToShow: number[];
  onPageChange: (page: number) => void;
};

class Paginator extends React.PureComponent<PropType> {
  static propTypes = {
    currentPage: PropTypes.number.isRequired,
    totalPages: PropTypes.number.isRequired,
    pagesToShow: PropTypes.arrayOf(PropTypes.number).isRequired,
    onPageChange: PropTypes.func.isRequired
  };
  render() {
    const { currentPage, totalPages, pagesToShow, onPageChange } = this.props;
    const isFirstPage = currentPage === 1,
      isLastPage = currentPage === totalPages;

    return (
      <nav aria-label="Page navigation" className="paginator">
        <ul className="pagination justify-content-center">
          <Page {...{ onPageChange, page: 1, isDisabled: isFirstPage }}>
            <span>First</span>
          </Page>

          <Page
            {...{
              onPageChange,
              page: currentPage - 1,
              isDisabled: isFirstPage
            }}
          >
            <span aria-hidden="true">&laquo;</span>
          </Page>

          {pagesToShow.map((page, index) => (
            <Page
              key={index}
              isDisabled={false}
              {...{ onPageChange, page, isActive: page === currentPage }}
            />
          ))}

          <Page
            {...{ onPageChange, page: currentPage + 1, isDisabled: isLastPage }}
          >
            <span aria-hidden="true">&raquo;</span>
          </Page>
          <Page {...{ onPageChange, page: totalPages, isDisabled: isLastPage }}>
            <span>Last</span>
          </Page>
        </ul>
      </nav>
    );
  }
}

export default Paginator;
