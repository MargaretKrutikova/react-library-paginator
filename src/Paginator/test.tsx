import * as React from 'react';
import * as Enzyme from 'enzyme';
import Paginator from '.';
import Page from '../Page';

describe('Paginator', () => {
  const testProps = {
    currentPage: 4,
    totalPages: 10,
    pagesToShow: [1, 2, 3, 4],
    onPageChange: jest.fn(),
    useBootstrapClasses: false
  };

  it(' renders all pages to show with active page', () => {
    const wrapper = Enzyme.shallow(<Paginator {...testProps} />);

    const createPageElement = (page: number, isActive: boolean) => (
      <Page
        onPageChange={testProps.onPageChange}
        useBootstrapClasses={testProps.useBootstrapClasses}
        isActive={isActive}
        isDisabled={false}
        page={page}
      />
    );
    const pagesToShowChildren = testProps.pagesToShow.map(page =>
      createPageElement(page, page === testProps.currentPage)
    );
    expect(wrapper.containsAllMatchingElements(pagesToShowChildren)).toBe(true);
  });

  it(' matches snapshot', () => {
    const wrapper = Enzyme.shallow(<Paginator {...testProps} />);

    expect(wrapper).toMatchSnapshot();
  });
});
