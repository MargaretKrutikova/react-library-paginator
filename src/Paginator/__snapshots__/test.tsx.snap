// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paginator  matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Paginator
    classes={Object {}}
    currentPage={4}
    navigation={Object {}}
    onPageChange={[MockFunction]}
    pagesToShow={
      Array [
        1,
        2,
        3,
        4,
      ]
    }
    styles={Object {}}
    totalPages={10}
    useBootstrapClasses={false}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "aria-label": "Page navigation",
      "children": <ul
        className=""
        style={undefined}
      >
        <Page
          classes={Object {}}
          isDisabled={false}
          onPageChange={[MockFunction]}
          page={1}
          styles={Object {}}
          useBootstrapClasses={false}
        >
          <span>
            First
          </span>
        </Page>
        <Page
          classes={Object {}}
          isDisabled={false}
          onPageChange={[MockFunction]}
          page={3}
          styles={Object {}}
          useBootstrapClasses={false}
        >
          <span>
            ←
          </span>
        </Page>
        <Page
          classes={Object {}}
          isActive={false}
          isDisabled={false}
          onPageChange={[MockFunction]}
          page={1}
          styles={Object {}}
          useBootstrapClasses={false}
        />
        <Page
          classes={Object {}}
          isActive={false}
          isDisabled={false}
          onPageChange={[MockFunction]}
          page={2}
          styles={Object {}}
          useBootstrapClasses={false}
        />
        <Page
          classes={Object {}}
          isActive={false}
          isDisabled={false}
          onPageChange={[MockFunction]}
          page={3}
          styles={Object {}}
          useBootstrapClasses={false}
        />
        <Page
          classes={Object {}}
          isActive={true}
          isDisabled={false}
          onPageChange={[MockFunction]}
          page={4}
          styles={Object {}}
          useBootstrapClasses={false}
        />
        <Page
          classes={Object {}}
          isDisabled={false}
          onPageChange={[MockFunction]}
          page={5}
          styles={Object {}}
          useBootstrapClasses={false}
        >
          <span>
            →
          </span>
        </Page>
        <Page
          classes={Object {}}
          isDisabled={false}
          onPageChange={[MockFunction]}
          page={10}
          styles={Object {}}
          useBootstrapClasses={false}
        >
          <span>
            Last
          </span>
        </Page>
      </ul>,
      "className": "",
      "style": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Page
            classes={Object {}}
            isDisabled={false}
            onPageChange={[MockFunction]}
            page={1}
            styles={Object {}}
            useBootstrapClasses={false}
          >
            <span>
              First
            </span>
          </Page>,
          <Page
            classes={Object {}}
            isDisabled={false}
            onPageChange={[MockFunction]}
            page={3}
            styles={Object {}}
            useBootstrapClasses={false}
          >
            <span>
              ←
            </span>
          </Page>,
          Array [
            <Page
              classes={Object {}}
              isActive={false}
              isDisabled={false}
              onPageChange={[MockFunction]}
              page={1}
              styles={Object {}}
              useBootstrapClasses={false}
            />,
            <Page
              classes={Object {}}
              isActive={false}
              isDisabled={false}
              onPageChange={[MockFunction]}
              page={2}
              styles={Object {}}
              useBootstrapClasses={false}
            />,
            <Page
              classes={Object {}}
              isActive={false}
              isDisabled={false}
              onPageChange={[MockFunction]}
              page={3}
              styles={Object {}}
              useBootstrapClasses={false}
            />,
            <Page
              classes={Object {}}
              isActive={true}
              isDisabled={false}
              onPageChange={[MockFunction]}
              page={4}
              styles={Object {}}
              useBootstrapClasses={false}
            />,
          ],
          <Page
            classes={Object {}}
            isDisabled={false}
            onPageChange={[MockFunction]}
            page={5}
            styles={Object {}}
            useBootstrapClasses={false}
          >
            <span>
              →
            </span>
          </Page>,
          <Page
            classes={Object {}}
            isDisabled={false}
            onPageChange={[MockFunction]}
            page={10}
            styles={Object {}}
            useBootstrapClasses={false}
          >
            <span>
              Last
            </span>
          </Page>,
        ],
        "className": "",
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <span>
              First
            </span>,
            "classes": Object {},
            "isDisabled": false,
            "onPageChange": [MockFunction],
            "page": 1,
            "styles": Object {},
            "useBootstrapClasses": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "First",
            },
            "ref": null,
            "rendered": "First",
            "type": "span",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <span>
              ←
            </span>,
            "classes": Object {},
            "isDisabled": false,
            "onPageChange": [MockFunction],
            "page": 3,
            "styles": Object {},
            "useBootstrapClasses": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "←",
            },
            "ref": null,
            "rendered": "←",
            "type": "span",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "class",
          "props": Object {
            "classes": Object {},
            "isActive": false,
            "isDisabled": false,
            "onPageChange": [MockFunction],
            "page": 1,
            "styles": Object {},
            "useBootstrapClasses": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "classes": Object {},
            "isActive": false,
            "isDisabled": false,
            "onPageChange": [MockFunction],
            "page": 2,
            "styles": Object {},
            "useBootstrapClasses": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "class",
          "props": Object {
            "classes": Object {},
            "isActive": false,
            "isDisabled": false,
            "onPageChange": [MockFunction],
            "page": 3,
            "styles": Object {},
            "useBootstrapClasses": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "class",
          "props": Object {
            "classes": Object {},
            "isActive": true,
            "isDisabled": false,
            "onPageChange": [MockFunction],
            "page": 4,
            "styles": Object {},
            "useBootstrapClasses": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <span>
              →
            </span>,
            "classes": Object {},
            "isDisabled": false,
            "onPageChange": [MockFunction],
            "page": 5,
            "styles": Object {},
            "useBootstrapClasses": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "→",
            },
            "ref": null,
            "rendered": "→",
            "type": "span",
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <span>
              Last
            </span>,
            "classes": Object {},
            "isDisabled": false,
            "onPageChange": [MockFunction],
            "page": 10,
            "styles": Object {},
            "useBootstrapClasses": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Last",
            },
            "ref": null,
            "rendered": "Last",
            "type": "span",
          },
          "type": [Function],
        },
      ],
      "type": "ul",
    },
    "type": "nav",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "aria-label": "Page navigation",
        "children": <ul
          className=""
          style={undefined}
        >
          <Page
            classes={Object {}}
            isDisabled={false}
            onPageChange={[MockFunction]}
            page={1}
            styles={Object {}}
            useBootstrapClasses={false}
          >
            <span>
              First
            </span>
          </Page>
          <Page
            classes={Object {}}
            isDisabled={false}
            onPageChange={[MockFunction]}
            page={3}
            styles={Object {}}
            useBootstrapClasses={false}
          >
            <span>
              ←
            </span>
          </Page>
          <Page
            classes={Object {}}
            isActive={false}
            isDisabled={false}
            onPageChange={[MockFunction]}
            page={1}
            styles={Object {}}
            useBootstrapClasses={false}
          />
          <Page
            classes={Object {}}
            isActive={false}
            isDisabled={false}
            onPageChange={[MockFunction]}
            page={2}
            styles={Object {}}
            useBootstrapClasses={false}
          />
          <Page
            classes={Object {}}
            isActive={false}
            isDisabled={false}
            onPageChange={[MockFunction]}
            page={3}
            styles={Object {}}
            useBootstrapClasses={false}
          />
          <Page
            classes={Object {}}
            isActive={true}
            isDisabled={false}
            onPageChange={[MockFunction]}
            page={4}
            styles={Object {}}
            useBootstrapClasses={false}
          />
          <Page
            classes={Object {}}
            isDisabled={false}
            onPageChange={[MockFunction]}
            page={5}
            styles={Object {}}
            useBootstrapClasses={false}
          >
            <span>
              →
            </span>
          </Page>
          <Page
            classes={Object {}}
            isDisabled={false}
            onPageChange={[MockFunction]}
            page={10}
            styles={Object {}}
            useBootstrapClasses={false}
          >
            <span>
              Last
            </span>
          </Page>
        </ul>,
        "className": "",
        "style": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <Page
              classes={Object {}}
              isDisabled={false}
              onPageChange={[MockFunction]}
              page={1}
              styles={Object {}}
              useBootstrapClasses={false}
            >
              <span>
                First
              </span>
            </Page>,
            <Page
              classes={Object {}}
              isDisabled={false}
              onPageChange={[MockFunction]}
              page={3}
              styles={Object {}}
              useBootstrapClasses={false}
            >
              <span>
                ←
              </span>
            </Page>,
            Array [
              <Page
                classes={Object {}}
                isActive={false}
                isDisabled={false}
                onPageChange={[MockFunction]}
                page={1}
                styles={Object {}}
                useBootstrapClasses={false}
              />,
              <Page
                classes={Object {}}
                isActive={false}
                isDisabled={false}
                onPageChange={[MockFunction]}
                page={2}
                styles={Object {}}
                useBootstrapClasses={false}
              />,
              <Page
                classes={Object {}}
                isActive={false}
                isDisabled={false}
                onPageChange={[MockFunction]}
                page={3}
                styles={Object {}}
                useBootstrapClasses={false}
              />,
              <Page
                classes={Object {}}
                isActive={true}
                isDisabled={false}
                onPageChange={[MockFunction]}
                page={4}
                styles={Object {}}
                useBootstrapClasses={false}
              />,
            ],
            <Page
              classes={Object {}}
              isDisabled={false}
              onPageChange={[MockFunction]}
              page={5}
              styles={Object {}}
              useBootstrapClasses={false}
            >
              <span>
                →
              </span>
            </Page>,
            <Page
              classes={Object {}}
              isDisabled={false}
              onPageChange={[MockFunction]}
              page={10}
              styles={Object {}}
              useBootstrapClasses={false}
            >
              <span>
                Last
              </span>
            </Page>,
          ],
          "className": "",
          "style": undefined,
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span>
                First
              </span>,
              "classes": Object {},
              "isDisabled": false,
              "onPageChange": [MockFunction],
              "page": 1,
              "styles": Object {},
              "useBootstrapClasses": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "First",
              },
              "ref": null,
              "rendered": "First",
              "type": "span",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span>
                ←
              </span>,
              "classes": Object {},
              "isDisabled": false,
              "onPageChange": [MockFunction],
              "page": 3,
              "styles": Object {},
              "useBootstrapClasses": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "←",
              },
              "ref": null,
              "rendered": "←",
              "type": "span",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "0",
            "nodeType": "class",
            "props": Object {
              "classes": Object {},
              "isActive": false,
              "isDisabled": false,
              "onPageChange": [MockFunction],
              "page": 1,
              "styles": Object {},
              "useBootstrapClasses": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "1",
            "nodeType": "class",
            "props": Object {
              "classes": Object {},
              "isActive": false,
              "isDisabled": false,
              "onPageChange": [MockFunction],
              "page": 2,
              "styles": Object {},
              "useBootstrapClasses": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "2",
            "nodeType": "class",
            "props": Object {
              "classes": Object {},
              "isActive": false,
              "isDisabled": false,
              "onPageChange": [MockFunction],
              "page": 3,
              "styles": Object {},
              "useBootstrapClasses": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "3",
            "nodeType": "class",
            "props": Object {
              "classes": Object {},
              "isActive": true,
              "isDisabled": false,
              "onPageChange": [MockFunction],
              "page": 4,
              "styles": Object {},
              "useBootstrapClasses": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span>
                →
              </span>,
              "classes": Object {},
              "isDisabled": false,
              "onPageChange": [MockFunction],
              "page": 5,
              "styles": Object {},
              "useBootstrapClasses": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "→",
              },
              "ref": null,
              "rendered": "→",
              "type": "span",
            },
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "children": <span>
                Last
              </span>,
              "classes": Object {},
              "isDisabled": false,
              "onPageChange": [MockFunction],
              "page": 10,
              "styles": Object {},
              "useBootstrapClasses": false,
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Last",
              },
              "ref": null,
              "rendered": "Last",
              "type": "span",
            },
            "type": [Function],
          },
        ],
        "type": "ul",
      },
      "type": "nav",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
